### PR TITLE
🔧 Integrate Shellcheck

### PR DESCRIPTION
**Pull request by sdwilsh**
**What this PR does / why we need it**:
This changes integrates `shellcheck` into the standard lint flow.  Along the way, it also fixes or suppresses any existing issues that exist.

### CODE REVIEW

File: .github/test.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 while [[ $count -ne 0 ]] ; do
     echo "Checking if kubeconfig is present"
     ./kairos get-kubeconfig > kube
-    cat kube | grep "http"
+    grep "http" < kube
```
Comment (Itxaka) [inline .github/test.sh:7]: already in love with this :D

**Review by Itxaka — COMMENTED** [review]
looks good, would like to still check manually to see if behaviour is the same, althougth looks good to me

**Review by mauromorales — DISMISSED** [review]
very cool :sunglasses:

**Review by mudler — APPROVED** [review]
Super nice!

**Review by Itxaka — APPROVED** [review]

File: scripts/start_vm_vbox.sh [hunk 2]
```diff
@@ -49,15 +55,30 @@ sleep 10
 
 set +e
 
+# FIXME: SC3006 (warning): In POSIX sh, standalone ((..)) is undefined.
+# https://www.shellcheck.net/wiki/SC3006
+# shellcheck disable=SC3006
```
[the remark this answers is not in the document]
Comment (sdwilsh) [reply scripts/start_vm_vbox.sh:60]: I suspect it works, so we could, but there's also a not-difficult fix suggested.  I suspect this is in shell, not bash, for OSX, and I don't have any way to test it, which is why I left it like I did.

**Pull request merged**